src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php

```diff
@@ -255,7 +255,8 @@ private function parseResponseList(ListShape $shape, string $input, bool $requir
             ]), $shape);
         }
 
-        return strtr($required ? '$this->FUNCTION_NAME(INPUT)' : 'empty(INPUT) ? [] : $this->FUNCTION_NAME(INPUT)', [
+        return strtr($required ? '$this->FUNCTION_NAME(INPUT)' : 'empty(INPUT) ? EMPTY : $this->FUNCTION_NAME(INPUT)', [
+            'EMPTY' => '$data' === substr($input, 0, 5) ? '[]' : 'null',
```

Comment:
this test on `$data` sounds really dubious

Reply:
Indeed.